### ISSUE TITLE
[Feature Request] Improve output of `--appimage-version` by adding date and commit id

### ISSUE THREAD

**Issue by KurtPfeifle** [AppImage/AppImageKit#533]
When running an AppImage with *`"--appimage-version"`*, the only info which currently prints could be:

     Version: continuous

Compare this to running *`"appimagetool --version"`*. This prints:

     Version: continuous-1-g63485ed

In case a user or one of the developers wants to debug something (or help a user with a problem), the second version is slightly more useful. So the same info should added to the output of *`"--appimage-version"`*.

However, even better would be to also add the *date* of the respective commit, for example like the following:

<pre>
./HelloWorld-x86_64.AppImage --appimage-version
  Version: continuous-abcd9876
  Date: 2017-11-20  11:13:56 UTC
</pre>

This way, a user would only have to memorize *"To have this bug fixed or have that feature added, I need a version later than Nov 20th, 2017"*.

**Comment by probonopd** [comment]
@TheAssassin has a much better version string in AppImageUpdate, maybe he can apply it here, too.

**Comment by KurtPfeifle** [comment]
After looking at version string in AppImageUpdate... yes, this should possibly be it:

     AppImageUpdate version 1-alpha (commit 71e2d90), build 121 built on 2017-11-07 19:41:18 UTC

However, for the use case of automatic parsing of output strings for specific info (maybe via shell scripts using `grep`, `awk` and co.), a *structured* output would be even better:

     AppName: AppImageUpdate
     Version: 1-alpha
     Commit: 71e2d90
     Build: 121
     BuildDate: 2017-11-07
     BuildTime: 19:41:18 UTC
     Compiler: LLVM version bla-blubb
     [....]

or, for an even more polished look:

        AppName: AppImageUpdate
        Version: 1-alpha
         Commit: 71e2d90
          Build: 121
      BuildDate: 2017-11-07
      BuildTime: 19:41:18 UTC
       Compiler: LLVM version bla-blubb
          [....]

**Comment by probonopd** [comment]
In Germany we call this a "Luxusproblem" but yes, that last one does look polished.

**Comment by probonopd** [comment]
By the way, we said we want to use the Git terminology "tag" instead of "version".

**Comment by KurtPfeifle** [comment]
Not sure what you want to say with your last comment. Is it that output should rather be:

     AppName: AppImageUpdate
      GitTag: 1-alpha
      [....]

or similar ?

<sub>(I know you not only like KISS -- you also like polish...  And yes, it's a "Luxusproblem". For many people on the Indian sub continent or elsewhere in the world, our whole AppImage activities are an effort to solve "Luxusprobleme", isn't it?)</pre>

**Comment by KurtPfeifle** [comment]
How about

       AppName: AppImageUpdate
        GitTag: 1-alpha
     GitCommit: 71e2d90
       BuildNo: 121
     BuildDate: 2017-11-07
     BuildTime: 19:41:18 UTC
      Compiler: LLVM version bla-blubb
       [....]

?

**Comment by probonopd** [comment]
Yes, but I'd rather not use CamelCase here, but only Git commit, Git tag, Build date,... or Commit, Tag, Date,...

**Comment by KurtPfeifle** [comment]
CamelCase is easier to parse if using text tools.

**Comment by TheAssassin** [comment]
> @TheAssassin has a much better version string in AppImageUpdate, maybe he can apply it here, too.

See https://github.com/AppImage/AppImageKit/pull/528.
